### DISCUSSION TITLE
Have "." be syntactic sugar for "this."

### DISCUSSION THREAD

**Discussion by victorcassel** [dotnet/csharplang#848] [General]
It is not rare to see `this.` in source code, although it is not needed. 
Maybe there is a reason! It brings some readability/understandability when you immediately see that a call to the current object is going on...
So I suggest `.` to be an allowed short form of `this.`

(I know you can write .5 instead of 0.5 which could be a problem)

**Comment by iam3yal** [comment]
@victorcassel What some of us do is prefix fields with an underscore, I don't know whether shortening `this.` into `.` really makes the code more readable, I think the opposite. 

I said it before but I think people confuse readable with terser code.

**Comment by HaloFour** [comment]
Why not just omit `this` entirely and use the identifier name directly?  You only ever need `this` when disambiguating between a local identifier that shadows a class member.  As such the explicit use of `this` should already be relatively uncommon, certainly not common enough to justify another syntax for it.

**Comment by jnm2** [comment]
This might block other proposed syntax that starts things with `.`. I don't think this use case is worth taking that.

**Comment by victorcassel** [comment]
Probably `this.` is a stressed programmer's "leave behind" after having used intellisense to save typing. But ask tool makers like resharper - it is a common cleanup suggestion. 

First discuss and analyse the phenomenon from the user perspective, later from the language implementer perspective.

**Comment by Joe4evr** [comment]
> it is a common cleanup suggestion.

But there are also people who prefer to have `this.` explicitly in every case. You can even set VS to tell/warn/error now when *not* specified, even when it's unambiguous.

Making code shorter != making code more readable. If it were, you might as well remove half the letters from your function names.

**Comment by theunrepentantgeek** [comment]
@Joe4evr There's real science behind readability, whether talking about code or regular prose. At the risk of over generalising, additional words take longer to read - as do misspelled ones.

**Comment by victorcassel** [comment]
@theunrepentantgeek What you say seems be a good argument against `this.`

**Comment by theunrepentantgeek** [comment]
Generally, I'm in favour of removing `this.` when its not absolutely required. 

In my experience, the rarity of `this.` tends to make it stick out like a sore thumb, calling attention to that piece of code. This unfortunately means that when I read code where `this.` is used heavily, it reads as though it's shouting all the time. #personalexperiences

**Comment by MhAllan** [comment]
Beside readability, You do need "this" keywords to filter the intellisense list while typing, especially when you have many imported namespaces, the "." suggestion is not readable, and will bother you if you are typing and missing a semicolon in the previous line.

**Comment by biohazard999** [comment]
You need this. to invoke an extension method on the current instance, so if you remove it, you need to use `MyStaticClass.DoStuff(this);` instead.
